File: Algorithms/Python/121.BestTimetoBuyandSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = len(prices)
        minp = float("inf")
        maxp = 0
        for i in range(l):
            if prices[i]<minp:
                minp = prices[i]
            else:
                maxp = max(maxp, prices[i]-minp)
        return maxp

File: Algorithms/Java/121.BestTimetoBuyandSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int min = Integer.MAX_VALUE;
        
        int l = prices.length;
        
        for (int i=0;i<l;i++){
            if (prices[i]<min){
                min = prices[i];
            }else{max=Math.max(max,prices[i]-min);}
        }
        return max;
    }
}